tests/testStreamOpen.cpp

```diff
@@ -444,3 +444,49 @@ TEST_F(StreamOpenOutput, OutputForOpenSLESPerformanceModeNoneGetBufferSizeInFram
     EXPECT_GT(mStream->getBufferSizeInFrames(), 0);
     ASSERT_TRUE(closeStream());
 }
+
+TEST_F(StreamOpenOutput, OpenSLESOutputReturnsMmapNotUsed){
+    mBuilder.setAudioApi(AudioApi::OpenSLES);
+    ASSERT_TRUE(openStream());
+    EXPECT_FALSE(mStream->isMMapUsed());
+    ASSERT_TRUE(closeStream());
+}
+
+TEST_F(StreamOpenInput, OpenSLESInputReturnsMmapNotUsed){
+    mBuilder.setAudioApi(AudioApi::OpenSLES);
+    mBuilder.setDirection(Direction::Input);
+    ASSERT_TRUE(openStream());
+    EXPECT_FALSE(mStream->isMMapUsed());
+    ASSERT_TRUE(closeStream());
+}
+
+TEST_F(StreamOpenOutput, OutputMmapUsedShouldMeanItsSupported){
+    ASSERT_TRUE(openStream());
+    if (mStream->isMMapUsed()){
+        EXPECT_TRUE(mBuilder.willUseMMap());
+    }
+    ASSERT_TRUE(closeStream());
+}
+
+TEST_F(StreamOpenInput, InputMmapUsedShouldMeanItsSupported){
+    mBuilder.setDirection(Direction::Input);
+    ASSERT_TRUE(openStream());
+    if (mStream->isMMapUsed()){
+        EXPECT_TRUE(mBuilder.willUseMMap());
+    }
+    ASSERT_TRUE(closeStream());
+}
+
+TEST_F(StreamOpenOutput, OutputMmapExclusiveSupportedShouldMeanItsSupported){
+    mBuilder.setDirection(Direction::Input);
+    if (mBuilder.willUseExclusiveMMap()){
```

Comment:
These tests that expect MMAP are not valid because the code does not make predictions about what will be used. The code is only reporting device support. Also some devices only support MMAP for output on headsets and not on other devices.